src/panel/cosmos.decorator.tsx

```diff
@@ -23,12 +23,32 @@ const FixtureStyle = createGlobalStyle`
   }
 `;
 
-export const ThemeDecorator: FC = ({ children, ...props }) => (
-  <ThemeProvider {...props} theme={theme}>
-    <GlobalStyle />
-    {children}
-  </ThemeProvider>
-);
+const ThemeToggle = styled.button`
+  position: fixed;
+  z-index: 1000;
+  bottom: 0;
+  right: 0;
+  top: initial !important;
+  background: #202020;
+  color: #fff;
+  padding: 10px;
+`;
+
+export const ThemeDecorator: FC = ({ children, ...props }) => {
+  const [theme, setTheme] = useState("dark");
+
+  return (
+    <ThemeProvider {...props} theme={theme === "dark" ? darkTheme : lightTheme}>
+      <ThemeToggle
+        onClick={() => setTheme(theme === "dark" ? "light" : "dark")}
```

Comment:
Nice, this is only for cosmos, isn't it? In the extension we are using chrome themes to detect if it's dark or white.

Reply:
Yeah, exactly!

Reply:
Potentially worth extracting this to a callback for some minor perf improvements

Reply:
@wgolledge thanks for the suggestion! Updated here: https://github.com/FormidableLabs/urql-devtools/pull/358/commits/54582a55c3f98a479a4ea413eb2e521564640517